src/action.ts

```diff
@@ -78,50 +78,47 @@ export const runAction = async (space): Promise<void> => {
     },
   };
 
-  if (CREATE_CDA_TOKEN) {
-    core.startGroup(`Ephemeral CDA token creation`)
-
-    const branchName = branchNames.headRef;
-    const keyName = `ephemeral-token-${branchName}`;
+  const branchName = branchNames.headRef;
+  const tokenKeyName = `ephemeral-token-${branchName}`;
 
+  if (CREATE_CDA_TOKEN) {
     const spaceKeys = await space.getApiKeys();
 
-    const exists = spaceKeys.items.some(item => item.name === keyName)
+    const exists = spaceKeys.items.some(item => item.name === tokenKeyName)
 
     if (exists) {
-      Logger.log(`CDA token ${keyName} is already created`);
+      Logger.log(`CDA token ${tokenKeyName} is already created`);
     } else {
-      Logger.log(`Creating new CDA token "${keyName}" for ephemeral environment "${environmentId}"...`);
+      Logger.log(`Creating new CDA token "${tokenKeyName}" for ephemeral environment "${environmentId}"...`);
 
       try {
         const key = await space.createApiKey({
-          name: keyName,
+          name: tokenKeyName,
           environments: [newEnv],
         })
 
-        // set token as secret just in case
-        core.setSecret(key.accessToken)
-      
-        core.setOutput(
-          "cda_token",
-          key.accessToken
-        );
         Logger.success("CDA token has been created");
       } catch(err) {
         Logger.warn("unable to create ephemeral token");
         Logger.verbose(err)
       }
     }
-
-    core.endGroup()
   }
 
   Logger.verbose("Update API Keys to allow access to new environment");
 
   const { items: keys } = await space.getApiKeys();
   await Promise.all(
     keys.map((key) => {
-      Logger.verbose(`Updating: "${key.sys.id}"`);
+      // put token value on every action run
+      // helpful in case the first run failed and the "Write Comment" step was not reached
+      if (key.name == tokenKeyName) {
+        Logger.verbose("debug: setting ephemeral token value to ouputs");
+        core.setOutput("cda_token", key.accessToken);
+        core.setSecret(key.accessToken) // set token as a secret after you've put it in the output!
```

Comment:
this was the culprit.
setting token as a secret before putting it to output resulted in... setting empty value?
setting it afterwards works exactly as it should 
```
Run actions/github-script@v6
  with:
    github-token: ***
    script: const output = `
  - See changes to the [sandbox-fix][1] environment on Contentful.
  - CDA token created \`***\`.
  [1]: https://app.contentful.com/spaces/***/environments/sandbox-fix
  `;
  github.rest.issues.createComment({
    issue_number: context.issue.number,
    owner: context.repo.owner,
    repo: context.repo.repo,
    body: output
  })
    debug: false
    user-agent: actions/github-script
    result-encoding: json
    retries: 0
    retry-exempt-status-codes: 400,401,403,404,422
```

Reply:
@jorgedavila25 just in case, it makes sense to "mark" newly created CDA token as a secret. 

Honestly, temporary CDA (read-only) token that is tied to specific ephemeral env might not be _that big of a secret_, but still, just in case, lets mark this token as a secret. So if somehow logs are going to leak or hackers will get access to our logs they will not see token value – it will be masked with `*****`. I am speculating of course.

The error I've made previously – I did set secret with `core.setSecret("cda_token_value")` _before_ I have put it in a special storage `outputs`. `outputs` content is available on the next workflow step. 
That resulted in writing empty token value in `outputs`. Consequently in the next workflow step token value was just empty.

The correct approach is to store token value in `outputs` _first_, then mark it as a secret with `core.setSecret()`
and this is what that PR fixes.

Reply:
@slash3b ah - i really appreciate the explanation. makes sense. thanks and good stuff!

Reply:
so marking it secret doesn't allow it to be added to output because, "oh hey, that's a secret key."

fascinating.